Added processing of the chronica_disabled flag

Gromobedg opened the pull request.

platinumthinker commented on src/chronica.erl:
```diff
@@ -37,7 +35,9 @@ register() ->
     end.
 
--spec register(atom()) -> no_return().
+-spec register(
```
Спеки лучше писать в одну строку, особенно если влазят в одну строку. Если не влазят то выносить отделные типы и уже далее если не влазит то выносить на разные строки

platinumthinker commented on src/chronica_disk_log_backend.erl:
```diff
@@ -181,24 +175,17 @@ proc([ Dir | Tail], ClearPath) -> proc(Tail, ClearPath ++ [Dir]);
-            end, [], Apps),
-
+            end, [], Apps
+        ),
```
Какая логика данного правила?

platinumthinker commented on src/chronica_error_logger_handler.erl:
```diff
@@ -14,29 +14,29 @@
+        warning_msg -> ?P_WARNING;
+        warning_report -> ?P_WARNING;
+        _ -> ?P_INFO
+    end,
```
Не лучше ли вынести этот case в отдельную функцию?

platinumthinker commented on src/chronica_file_backend.erl:
```diff
@@ -194,22 +195,17 @@ proc([ Dir | Tail], ClearPath) -> proc(Tail, ClearPath ++ [Dir]);
-        catch
-            _:_ -> "build_info is unavailable\n"
-        end,
+    Str = try get_application_info() catch _:_ -> "build_info is unavailable\n" end,
```
А здесь наоборот пошли от обратного и слепили всё в одну строку. Может лучше всё же в едином стиле делать?

platinumthinker commented on src/chronica_file_backend.erl:
```diff
@@ -237,18 +236,22 @@ get_application_build_info(AppName) ->
-            _:_ -> 0
-        end,
+    try
+        Module:build_info(),
```
Здесь Module:build_info вызывается для побочного действия - подгрузки модуля. Лучше делать это явно через code:load_file(Module). Пишу чтобы не забылось. В этом комите это править не стоит

platinumthinker commented on src/chronica_format_creator.erl:
```diff
@@ -97,114 +102,136 @@ priority_color(FunColor) ->
-    make_format_function_ast(Res, BinaryStr ++ NewBinaryStr ++ theend(Res),
-                             StrAST ++ NewStrAST, FunColor, FlagColor);
+    make_format_function_ast(
+        Res, BinaryStr ++ NewBinaryStr ++ theend(Res), StrAST ++ NewStrAST, FunColor, FlagColor
```
Лучше разбить на отдельные строки и писать например так OutAST = lists:append([StrAST, NewStrAST]), а потом использовать OutAST

platinumthinker replied:
И судя по всему если ты не заметил строчка 
```
make_format_function_ast(
                            StrAST ++ NewStrAST, FunColor, FlagColor);
       Res, BinaryStr ++ NewBinaryStr ++ theend(Res), StrAST ++ NewStrAST, FunColor, FlagColor
```
Повторяется несколько раз. Может это как то исправить чтобы поторяемости не было?

platinumthinker commented on src/chronica_gen_backend.erl:
```diff
@@ -126,47 +142,50 @@ handle_cast(_Unknown, State) ->
     end;
-
-handle_call({close, Output, Handle, Timeout}, _From, State = #s{files = Files}) when Timeout =< 0 ->
+handle_call({close, Output, Handle, Timeout}, _From, State) when Timeout =< 0 ->
```
Поменялась функциональность! Раньше сюда бы что то не сматчилось, но теперь сматчится с любым State и потом внутри функции упадёт.

Gromobedg replied:
Ну это бред, это handle_call, откуда здесь другой state появится

s-kostyaev replied:
Бывало и такое после неудачного рефакторинга.

platinumthinker commented on src/chronica_gen_backend.erl:
```diff
@@ -126,47 +142,50 @@ handle_cast(_Unknown, State) ->
     end;
-handle_call({close, Output, Handle, Timeout}, _From, State = #s{closing_flows = ClosingFlows}) when Timeout > 0 ->
+handle_call({close, Output, Handle, Timeout}, _From, State) when Timeout > 0 ->
+    #s{closing_flows = ClosingFlows} = State,
```
Аналогично предыдущему коментарию

platinumthinker commented on src/chronica_internal_logger.erl:
```diff
@@ -56,23 +62,20 @@ init(Params) ->
+        ?MPARAM(handle)
+    catch
+        _:_ -> null
+    end,
```
На будущее (не для этого комита) - стоит избавиться от case и внести disk_log:sync\close в try catch

platinumthinker commented on src/chronica_internal_logger.erl:
```diff
@@ -93,62 +96,84 @@ log(P, Time, F, A, Module, Func, Line) ->
--spec log_error(Time::term(), string(), [any()], atom(), string(), integer()) -> ok.
-
+-spec log_error(
+    Time::term(),
```
Спек стоит оставить написанным в одну строку.

platinumthinker commented on src/chronica_internal_logger.erl:
```diff
@@ -93,62 +96,84 @@ log(P, Time, F, A, Module, Func, Line) ->
-                                   {type, wrap},
-                                   {size, Sizes}]) of
+    FlagOpen = disk_log:open([
+        {name, LogName}, {file, Filename}, {format, external}, {type, wrap}, {size, Sizes}
```
Где то в аналогичной кострукции ты разбивал на каждый tuple в новой строке. Я думаю стоит оставаться в той же концепции

platinumthinker commented on src/chronica_output.erl:
```diff
@@ -14,37 +14,40 @@
         Mk = erlang:element(3, NowTime),
-        put_to_outputs({{Date, Time, Mk}, Priority, Module, Pid, Line, File, Function, Format, Args}, default, Flow)
+        put_to_outputs(
+            {{Date, Time, Mk}, Priority, Module, Pid, Line, File, Function, Format, Args},
```
На будущее - число аргументов тут адское. Наверное тут поможет какая нибудь инкапсуляция, типа Place :: {Module, Line, File, Function} и потом работа с такими типами должна происходить в отдельных функциях

platinumthinker commented on src/chronica_parser.erl:
```diff
@@ -68,34 +63,27 @@ parse_control_sequence_token(_Input) ->
 
 parse_control_sequence_field_width([Char | Input], Token) when (Char >= $0 andalso Char =< $9) orelse Char =:= $+ orelse Char =:= $- ->
```
Здесь 136 символов - недоглядел

platinumthinker commented on src/chronica_starter.erl:
```diff
@@ -24,61 +24,55 @@
+    Priority :: debug | trace | info | warning | error,
+    Mask :: string(), % use &,*,?,|,!
+    FormatLog :: default | short | html,
+    ListManualStartApp :: [atom() | {application, atom()}]
```
Из спека стоит вынести типы - Priority, FormatLog, StartApp и тогда данная сточка будет легче читаться. Я думаю что такой спек даже описан в данном приложении

platinumthinker commented on src/chronica_tcp_con_backend.erl:
```diff
@@ -28,11 +35,12 @@ handle_close(Con) ->
     try
-        chronica_tcp_connection:write(erlang:binary_to_term(Con), chronica_format_creator:frame(Data))
+        chronica_tcp_connection:write(
+            erlang:binary_to_term(Con), chronica_format_creator:frame(Data)
```
Правильнее разнести на несколько независимых строк. Что то вроде
```
Con = erlang:binary_to_term(BinCon),
DataFrame = chronica_format_creator:frame(Data),
chronica_tcp_connection:write(Con, DataFrame),
```

platinumthinker commented on src/chronica_tcp_connection.erl:
```diff
@@ -56,129 +70,126 @@ close(Con, Reason) ->
-
+-spec handle_call(
+    Request :: term(),
+    From :: term(),
```
Предыдущая версия спека мне нравилась больше. Наверное можно её просто сдвинуть влево (много табов было)

platinumthinker commented on src/chronica_tcp_connection.erl:
```diff
@@ -56,129 +70,126 @@ close(Con, Reason) ->
+    {noreply, State :: #s{}, Timeout :: non_neg_integer()} |
+    {stop, Reason :: term(), State :: #s{}}.
+handle_cast(start_accept, State) ->
+    #s{listen_socket = ListenSocket, man_mon = ManMon, manager = Manager} = State,
```
Неэквивалентная запись - раньше с неправильным state оно бы попало в 
```
 handle_cast(_Msg, State) ->
     ?INT_ERR("Unhandled cast: ~10000p", [_Msg]),
     {noreply, State}.
```

platinumthinker commented on src/chronica_tcp_connection.erl:
```diff
@@ -56,129 +70,126 @@ close(Con, Reason) ->
-
--spec terminate(Reason :: term(), State :: #s{}) -> ok.
+-spec terminate(
+    Reason :: term(),
```
Спек в одну строку вполне раньше влазил

platinumthinker commented on src/chronica_tcp_server.erl:
```diff
@@ -53,18 +63,22 @@ get_listen_port() ->
         IP ->
-            case catch gen_tcp:listen(Port, [binary, {ifaddr, IP}, {packet, 0}, {active, true}, {nodelay, false}, {keepalive, true}, {backlog, ?DEFAULT_TCP_BACKLOG}, {reuseaddr, true}]) of
+            ParamsGenTCP = [
+                binary, {ifaddr, IP}, {packet, 0}, {active, true}, {nodelay, false},
```
Как мне кажется тут лучше по одному-двум таплам на строку делать

platinumthinker commented on src/pt_chronica.erl:
```diff
@@ -365,33 +370,41 @@ generate_module_iface_name(Module) ->
-                                              end, "", Format)),
+    EscapedFormat = lists:reverse(
+        lists:foldl(
+            fun ($~, Acc) -> [$~, $~|Acc];
```
Данный блок лучше вынести в функцию, тем более что он используются чуть выше

hurd54 commented:
Соглашусь, матчить стейт внутри генсервера - излишне.

Отправлено с iPhone

> 25 мая 2017 г., в 18:51, Nikita Roshupkin <notifications@github.com> написал(а):
> 
> @Gromobedg commented on this pull request.
> 
> In src/chronica_gen_backend.erl:
> 
> >              Error -> {reply, Error, State}
>          end
>      catch
>          _:E -> {reply, {error, {E, erlang:get_stacktrace()}}, State}
>      end;
> -
> -handle_call({close, Output, Handle, Timeout}, _From, State = #s{files = Files}) when Timeout =< 0 ->
> +handle_call({close, Output, Handle, Timeout}, _From, State) when Timeout =< 0 ->
> Ну это бред, это handle_call, откуда здесь другой state появится
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

Commit:
Added processing of the chronica_disabled flag

You can set it using environment flag export CHRONICA_DISABLED=<any word>
or erl_opts <chronica_disabled>

platinumthinker commented:
Pull request without feature branch
